packages/repocop/src/remediations/repository-06-topic-monitor-production.ts

```
@@ -16,49 +51,138 @@ export function getReposWithoutProductionTopic(
-	return tags.map((tag) => tag['gu:repo']).filter((r) => !!r) as string[];
+export function getGuRepoName(tag: AWSCloudformationTag): string | undefined {
+	return tag['gu:repo'];
 }
 
 export function getReposInProdWithoutProductionTopic(
 	reposWithoutProductionTopic: string[],
-	guReposWithProdCfnStacks: string[],
-) {
-	return reposWithoutProductionTopic.filter((repoName) =>
-		guReposWithProdCfnStacks.includes(repoName),
-	);
+	guRepoStacks: GuRepoStack[],
+): GuRepoStack[] {
+	return guRepoStacks.filter((stack) => {
+		const guRepoName: string = stack.guRepoName;
+		return reposWithoutProductionTopic.includes(guRepoName);
+	});
 }
 
-export async function findReposInProdWithoutProductionTopic(
+async function findReposInProdWithoutProductionTopic(
 	prisma: PrismaClient,
 	unarchivedRepos: github_repositories[],
 ) {
-	console.log('Discovering Cloudformation stacks with PROD or INFRA tags');
-	const reposWithoutProductionTopic: string[] =
-		getReposWithoutProductionTopic(unarchivedRepos);
+	console.log('Discovering Cloudformation stacks with PROD or INFRA tags.');
+
+	const repoNamesWithoutProductionTopic: string[] =
+		getRepoNamesWithoutProductionTopic(unarchivedRepos);
 	console.log(
-		`Found ${reposWithoutProductionTopic.length} repositories without a production or interactive topic`,
+		`Found ${repoNamesWithoutProductionTopic.length} repositories without a production or interactive topic.`,
 	);
 
-	const repoTagsWithProdCfnStacks = await findProdCfnStackTags(prisma);
+	const cfnStacksWithProdInfraTags: AWSCloudformationStack[] =
+		await findProdCfnStacks(prisma);
 
-	const guReposWithProdCfnStacks: string[] = getGuRepoNames(
-		repoTagsWithProdCfnStacks,
-	);
+	const guRepoStacks: GuRepoStack[] = cfnStacksWithProdInfraTags
+		.filter(
+			(stack: AWSCloudformationStack) =>
+				getGuRepoName(stack.tags) !== undefined,
+		)
+		.map((stack: AWSCloudformationStack) => {
+			const guRepoName = getGuRepoName(stack.tags) as string;
```

This cast will mask errors that'll only seen at runtime as `getGuRepoName` returns `string | undefined`. Checking if `guRepoName` is `undefined` would be more type-safe.